src/middlewares/authentication.js

```diff
@@ -78,14 +78,10 @@ export default {
     const { id } = request.user;
     const { slug } = request.params;
 
-    const articleResponse = await Articles.findOne({
-      where: { slug }
-    });
+    const articleResponse = await Articles.findOneArticle(slug);
 
     if (!articleResponse) throw new ApplicationError(404, 'Article not found');
-    const article = await Articles.findOne({
-      where: { authorId: id, slug }
-    });
+    const article = await Articles.findOneArticle(slug, { authorId: id });
 
     if (!article) throw new ApplicationError(403, 'Unauthorized Access. For author only');
```

Comment:
Please just do one find. Find an article that has the slug, if it doesn't exist, return 404, if the authorId is not equal to the user id, then return this 403. We are making too many requests and as we scale, the app will get slower. Thanks

Reply:
> Please just do one find. Find an article that has the slug, if it doesn't exist, return 404, if the authorId is not equal to the user id, then return this 403. We are making too many requests and as we scale, the app will get slower. Thanks

It checks if the followupId is a valid author id and also checks if any of the tags provided is invalid or is not concurrent with the predefined categories in our database.
It just checks to ensure that followupId and tags are valid.

Reply:
True, we should reduce database calls as much as possible.

Reply:
> Please just do one find. Find an article that has the slug, if it doesn't exist, return 404, if the authorId is not equal to the user id, then return this 403. We are making too many requests and as we scale, the app will get slower. Thanks

True, we should reduce database calls as much as possible.